### PR TITLE
Add demo for 2D steady-state adaptation

### LINKED ISSUE #93: 2D steady-state mesh adaptation demo
Current the demos stop before doing any mesh adaptation! The demo should use fixed point iteration.

### PR DESCRIPTION
**Pull request by jwallwork23**
The first of a series of new demos that actually use the mesh adaptation functionality.

Closes #93. Partially addresses #87.

### CODE REVIEW

**Comment by jwallwork23** [comment]
Let's discuss/review this at the meeting today.

File: test/Makefile [hunk 1]
```diff
@@ -2,7 +2,7 @@ all: run
 
 run:
 	@echo "Running all tests..."
-	@python3 -m pytest -v .
+	@python3 -m pytest -v --durations=20 .
```
Comment (stephankramer) [inline test/Makefile:5]: This looks like a useful option - TIL

**Review by stephankramer — APPROVED** [review]
That all looks really good - and very timely for the student projects that are just starting

**Comment by jwallwork23** [comment]
Thanks, will merge now and plan to get working on the subsequent demos in the coming days.

**Comment by jwallwork23** [comment]
Updated the website. You can find the new demo [here](https://pyroteus.github.io/demos/point_discharge2d-hessian.py.html).

**Pull request merged**